File: java/Queues/Queue.java
package Queues;

import Lists.DoubleLinkedList;
import definitions.Queues;

public class Queue<T> implements Queues<T> {

    // could have extended directly to DLL but want to limit the features like standard queue
    private DoubleLinkedList<T> list;

    public Queue() {
        this.list = new DoubleLinkedList<>();
    }


    @Override
    public void add(T val) {
        list.addLast(val);
    }

    @Override
    public T remove() {
        return list.removeFirst();
    }

    @Override
    public int size() {
        return list.size();
    }

    @Override
    public T peek() {
        return list.getFirst();
    }

    @Override
    public boolean isEmpty() {
        return list.isEmpty();
    }

    @Override
    public void clear() {
        list.clear();
    }
}

File: typescript/src/queues/queue.ts
import DoubleLinkedList from "../lists/double-linked-list.ts";

export default class Queue<T> {
  // could have extended directly to DLL but want to limit the features like standard queue
  private list: DoubleLinkedList<T>;

  constructor() {
    this.list = new DoubleLinkedList();
  }

  public add(val: T) {
    this.list.addLast(val);
  }

  public remove() {
    return this.list.removeFirst();
  }

  public peek() {
    return this.list.getFirst();
  }

  public get size() {
    return this.list.size;
  }

  public clear() {
    this.list.clear();
  }
}
